ctru-rs/src/test_runner.rs

```diff
@@ -0,0 +1,135 @@
+//! Custom test runner for building/running unit tests on the 3DS.
+
+extern crate test;
+
+use std::io;
+
+use test::{ColorConfig, Options, OutputFormat, RunIgnored, TestDescAndFn, TestFn, TestOpts};
+
+use crate::console::Console;
+use crate::gfx::Gfx;
+use crate::services::hid::{Hid, KeyPad};
+
+/// A custom runner to be used with `#[test_runner]`. This simple implementation
+/// runs all tests in series, "failing" on the first one to panic (really, the
+/// panic is just treated the same as any normal application panic).
+pub(crate) fn test_runner(tests: &[&TestDescAndFn]) {
+    crate::init();
+
+    let gfx = Gfx::default();
+    let hid = Hid::init().unwrap();
+
+    let mut top_screen = gfx.top_screen.borrow_mut();
+    top_screen.set_wide_mode(true);
+    let _console = Console::init(top_screen);
+
+    // Start printing from the top left
+    print!("\x1b[1;1H");
+
+    // TODO: it would be nice to have a way of specifying argv to make these
+    // configurable at runtime, but I can't figure out how to do it easily,
+    //  so for now, just hardcode everything.
+    let opts = TestOpts {
+        list: false,
+        filters: Vec::new(),
+        filter_exact: false,
+        // Forking is not supported
+        force_run_in_process: true,
+        exclude_should_panic: false,
+        run_ignored: RunIgnored::No,
+        run_tests: true,
+        // Benchmarks are not supported
+        bench_benchmarks: false,
+        logfile: None,
+        nocapture: false,
+        // TODO: color doesn't work because of TERM/TERMINFO.
+        // With RomFS we might be able to fake this out nicely...
+        color: ColorConfig::AlwaysColor,
+        format: OutputFormat::Pretty,
+        shuffle: false,
+        shuffle_seed: None,
+        // tweak values? This seems to work out of the box
+        test_threads: Some(3),
```

Comment:
I thought we disabled std thread spawning by breaking the pthread impl?
(I would like to keep std thread spawning, but also there's a good reason to avoid it)

Reply:
Yeah, I wasn't really expecting this to work right away... maybe it's safer to disable concurrency in the test runner at least until #45 is resolved?

Reply:
Okay, I figured it out. After `cargo update` this no longer seems to work like it did before, so I will swap to `None` instead. We can look at re-enabling after [#45](https://github.com/Meziu/ctru-rs/issues/45) is settled.